sdk/storage/Azure.Storage.Blobs.ChangeFeed/src/ChangeFeedFactory.cs

```diff
@@ -0,0 +1,191 @@
+            DateTimeOffset? startTime = default,
+            DateTimeOffset? endTime = default,
+            string continuation = default)
+        {
+            BlobContainerClient containerClient = blobServiceClient.GetBlobContainerClient(Constants.ChangeFeed.ChangeFeedContainerName);
+            DateTimeOffset lastConsumable;
+            Queue<string> years;
+            Queue<string> segments;
+            ChangeFeedCursor cursor = null;
+
+            // Create cursor
+            if (continuation != null)
+            {
+                cursor = JsonSerializer.Deserialize<ChangeFeedCursor>(continuation);
+                ValidateCursor(containerClient, cursor);
+                startTime = cursor.CurrentSegmentCursor.SegmentTime;
+                endTime = cursor.EndTime;
+            }
+            // Round start and end time if we are not using the cursor.
+            else
+            {
+                startTime = startTime.RoundDownToNearestHour();
+                endTime = endTime.RoundUpToNearestHour();
+            }
+
+            // Check if Change Feed has been abled for this account.
+            bool changeFeedContainerExists;
+
+            if (async)
+            {
+                changeFeedContainerExists = await containerClient.ExistsAsync().ConfigureAwait(false);
+            }
+            else
+            {
+                changeFeedContainerExists = containerClient.Exists();
+            }
+
+            if (!changeFeedContainerExists)
+            {
+                throw new ArgumentException("Change Feed hasn't been enabled on this account, or is currently being enabled.");
+            }
+
+            // Get last consumable
+            BlobClient blobClient = containerClient.GetBlobClient(Constants.ChangeFeed.MetaSegmentsPath);
+            BlobDownloadInfo blobDownloadInfo;
+            if (async)
+            {
+                blobDownloadInfo = await blobClient.DownloadAsync().ConfigureAwait(false);
+            }
+            else
+            {
+                blobDownloadInfo = blobClient.Download();
+            }
+
+            JsonDocument jsonMetaSegment;
+            if (async)
+            {
+                jsonMetaSegment = await JsonDocument.ParseAsync(blobDownloadInfo.Content).ConfigureAwait(false);
+            }
+            else
+            {
+                jsonMetaSegment = JsonDocument.Parse(blobDownloadInfo.Content);
+            }
+
+            lastConsumable = jsonMetaSegment.RootElement.GetProperty("lastConsumable").GetDateTimeOffset();
+
+            // Get year paths
+            years = await GetYearPaths(async, containerClient).ConfigureAwait(false);
+
+            // Dequeue any years that occur before start time
+            if (startTime.HasValue)
+            {
+                while (years.Count > 0
+                    && years.Peek().ToDateTimeOffset() < startTime.RoundDownToNearestYear())
+                {
+                    years.Dequeue();
```

Comment:
Is it better if we sort the `years[]` first accroding to the number.  "Blobs are listed in alphabetical order ", then year 996 will be listed after 2020, same for 9999 and 10000, although our code probably won't run that long for the bug to kick in.

Reply:
I don't think we need to be worried about it.